contracts/data/EnumerableMap.sol

```diff
@@ -35,11 +35,19 @@ library EnumerableMap {
         returns (address, address)
     {
         (bytes32 key, bytes32 value) = _at(map._inner, index);
+
         address addressKey;
+        address addressValue;
+
         assembly {
-            addressKey := mload(add(key, 20))
+            addressKey := and(0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF, key)
```

Comment:
This is not an elegant solution. 

I'll have a look at what else can be done.

Reply:
I don't know why there was an assembly block here.  All we need is a simple cast: `return (address(uint160(uint256(key))), address(uint160(uint256(value))));`